src/domain/file/store/selector/file.selector.ts

```diff
@@ -0,0 +1,18 @@
+import { createSelector } from 'reselect'
+import type { DeepReadonly } from 'superTypes'
+import type { AppState, FileById } from '../appState'
+import type { MediaType } from 'domain/common/type'
+import type { FileEntity } from 'domain/file/entity/file'
+
+export type FileDescriptor = {
+  readonly name: string
+  readonly size: number
+  readonly type: MediaType
+}
+
+export const getFiles: (state: DeepReadonly<AppState>) => FileDescriptor[] = createSelector(
+  (state: DeepReadonly<AppState>): FileById => state.file.byId,
+  (files: DeepReadonly<FileById>): FileDescriptor[] => [
+    ...files.map(({ name, size, type }: FileEntity) => ({ name, size, type })).toList()
```

Comment:
I think here we can avoid the use of the `immutable` `map` function that returns a new `OrderedMap` which weighs down the function for nothing. 
IMHO, It's better to first convert the `OrderedMap` into an JS `Array` and then `map` on it.
```suggestion
    files
      .toIndexedSeq()
      .toArray()
      .map(({ id, name, size, type }: FileEntity) => ({ id, name, size, type }))
```